README.md
# Women's Refuge Shielded Site #

Donate link: https://womensrefuge.org.nz/support-our-work/make-a-donation/

Wordpress Plugin to add the NZ Women's Refuge Shielded Site button to your site using the `[womens_refuge_shield]` shortcode or widget.

## Description ##

Now any website can be a place of refuge. If you need help but are worried about repercussions from a controlling or abusive partner, this is a safe way to find it. With The Women's Refuge Shielded Site, victims of domestic violence access a shielded portal through which they can contact us, find out how to make a plan to safely get out of a dangerous situation, learn how to stay safe online and get answers to questions about what comes next – all without fear of it showing up in their browser history for anyone else to find.

The Shielded Site is simple to add to any company website and is designed to work on desktop, tablet and mobile.

By including the Shielded Site tab on your website, you’ll be helping victims living with domestic violence to safely get help. We invite all New Zealand businesses – or anyone who manages their own website – to get involved.

**Note:** This plugin follows the example given on the [Shielded Site](https://shielded.co.nz/ "The Shielded Site Project") website, and therefore embeds a link to a hosted JavaScript file from their website. Additionally, clicking the Shielded Site image generated by the shortcode or widget from this app is *explicitly* designed to display another (external) websites' content over the top of your website.

## Frequently Asked Questions ##

### Is this an official Shielded Site plugin? ###

No. The plugin has been provided by the author, who wanted an easy way to include the Shielded Site features (https://shielded.co.nz/) on their WordPress websites, but could find no neat ways of doing so.

### How can I include the shielded logo on my site? ###

The plugin provides two options
1. You can insert the [womens_refuge_shield] shortcode where you want it on the page.
2. You can insert the Widget into where you wish to use it.

### What options are available? ###

You can control to look of the icon by changing the icon_size parameter (valid options: *large* (default), *small*, or *button*).

For the more technical users, you can also affect the ID of the inserted image using the `element_id` parameter, and the ID of the modal overlay using the `modal_id` parameter.

### Can I control the styling ###

Yes. The link image is inserted to your site inside a `<div>` element, with a class of `".wrss_container"`.

### Can I include more than one version on my page? ###

Yes. **However** you should note that only the last one on the page will activate when you click on it, due to a limitation from the code provided by https://shielded.co.nz

## Upgrade Notice ##

### 1.0.4 ###
* Tested against 6.1.1
* Code updated to align to [WordPress Coding Standards](https://developer.wordpress.org/coding-standards/wordpress-coding-standards)
* Change to screenshot to remove dev site name

### 1.0 ###
By including the Shielded Site tab on your website, you’ll be helping victims living with domestic violence to safely get help. Women's Refuge invite all New Zealand businesses – or anyone who manages their own website – to get involved.


changelog.md
# Changelog #

## 1.0.4 ##
* Tested against WordPress v6.1.1
* Code updated to align to [WordPress Coding Standards](https://developer.wordpress.org/coding-standards/wordpress-coding-standards)
* Change to screenshot to remove dev site name

## 1.0.3 ##
* Tested up to increased to 6.0.1
* Changes moved to `changelog.md`
* SVN deploy / Git Tag script created

## 1.0.2 ##
* Multi-lingual setup
* Assets created (Banner, Icon, and Screenshots)
* Tested against 5.7.1
* SVN config

## 1.0.1 ##
* Changes requested by WordPress approvals team
* Enqueue external scripts
* Stable Tag alignment.

## 1.0 ##
* Release to WordPress Plugin repository.

## 0.0.2 ##
* Extended features to allow a widget instead of just shortcode.

## 0.0.1 ##
* Initial version allowing embedding via shortcode.


deploy.sh
#! /bin/bash
# A modification of Sal Ferrarello's deploy script as found here: https://github.com/salcode/stop-emails/blob/master/deploy.sh

# main config
PLUGINSLUG="womens-refuge-shielded-site" # Used for main plugin filename, as well as SVN repo link.
CURRENTDIR=`pwd`
MAINFILE="$PLUGINSLUG.php" # this should be the name of your main php file in the wordpress plugin

# git config
GITPATH="$CURRENTDIR/" # this file should be in the base of your git repository

# svn config
SVNPATH="/tmp/$PLUGINSLUG" # path to a temp SVN repo. No trailing slash required and don't add trunk.
SVNURL="http://plugins.svn.wordpress.org/$PLUGINSLUG/" # Remote SVN repo on wordpress.org, with no trailing slash
SVNUSER=`grep "^Contributors:" $GITPATH/README.txt | awk -F' ' '{print $NF}'` # This won't work if there's more than one or a comma etc.
#SVNUSER="fychan66" # your svn username (uncomment & use instead if you have >1 contributor)


# Let's begin...
echo ".........................................."
echo
echo "Preparing to deploy wordpress plugin"
echo
echo ".........................................."
echo

# Check version in readme.txt is the same as plugin file after translating both to unix line breaks to work around grep's failure to identify mac line breaks
NEWVERSION1=`grep "^Stable tag:" $GITPATH/README.txt | awk -F' ' '{print $NF}'`
echo "README.txt version: $NEWVERSION1"
#echo "$GITPATH$MAINFILE"
NEWVERSION2=`grep "Version:" $GITPATH$MAINFILE | awk -F' ' '{print $NF}'`
echo "$MAINFILE version: $NEWVERSION2"

# Tweaked to check string equality, as I use semver versioning, not integers.
if [ "$NEWVERSION1" != "$NEWVERSION2" ]
    then
    echo "Version in README.txt & $MAINFILE don't match. Exiting....";
    read -n 1 -s -r -p "Press any key to continue";
    exit 1;
fi

echo "Versions match in readme.txt and $MAINFILE. Let's proceed..."

if git show-ref --tags --quiet --verify -- "refs/tags/$NEWVERSION1"
    then
		echo "Version $NEWVERSION1 already exists as git tag. Exiting....";
    read -n 1 -s -r -p "Press any key to continue";
		exit 1;
	else
		echo "Git version does not exist. Let's proceed..."
fi

cd $GITPATH
echo -e "Enter a commit message for this new version: \c"
read COMMITMSG
if [ "$COMMITMSG" = "" ]
    then
    echo "No commit message. Exiting....";
    read -n 1 -s -r -p "Press any key to continue"
fi

git commit -am "$COMMITMSG"

echo "Tagging new version in git"
git tag -a "$NEWVERSION1" -m "Tagging version $NEWVERSION1"

echo "Pushing latest commit to origin, with tags"
git push origin main
git push origin main --tags

echo
echo "Creating local copy of SVN repo ..."
svn co $SVNURL $SVNPATH

echo "Exporting the HEAD of master from git to the trunk of SVN"
git checkout-index -a -f --prefix=$SVNPATH/trunk/

echo "Ignoring github specific files and deployment script"
svn propset svn:ignore "deploy.sh
README.md
.wordpress-org
.gitattributes
.gitignore
.git
.distignore" "$SVNPATH/trunk/"

echo "Changing directory to SVN and committing to trunk"
cd $SVNPATH/trunk/
# Add all new files that are not set to be ignored
svn status | grep -v "^.[ \t]*\..*" | grep "^?" | awk '{print $2}' | xargs svn add
svn commit --username=$SVNUSER -m "$COMMITMSG"

echo "Creating new SVN tag & committing it"
cd $SVNPATH
svn copy trunk/ tags/$NEWVERSION1/
cd $SVNPATH/tags/$NEWVERSION1
svn commit --username=$SVNUSER -m "Tagging version $NEWVERSION1"

echo "Removing temporary directory $SVNPATH"
rm -fr $SVNPATH/

echo "*** FIN ***"
read -n 1 -s -r -p "Press any key to continue"


js/embed.js
var ds07o6pcmkorn=function(e){
  var t=this;
  t.modalID=e.modalID || "modal",
  t.left=e.left || 0,
  t.top=e.top || 0,
  t.icon=e.iconSize || "big",
  t.url="https://staticcdn.co.nz",
  t.smallIcon=t.url+"/embed/sm.png",
  t.bigIcon=t.url+"/embed/lg.png",
  t.closeIcon=t.url+"/embed/close.png",
  t.elementId=e.openElementId || "",
  t.appended=!1,
  t.btn=function(){
    return '<img id="frame-opener" style="cursor: pointer; z-index:  99999; -webkit-transition: all .5s; position: fixed; top:'+t.top+"px; left: "+t.left+'px; max-width: 60px" src="'+t.iconSize(t.icon)+'" alt="Enter the Shielded Site"/>'
  },
  t.frame='<div id="'+t.modalID+'" style="display:none; overflow: auto;  position: fixed; top: 0; left: 0; right: 0; bottom: 0; z-index: 99999; -webkit-overflow-scrolling: touch; background: rgba(0,0,0,0.5); zoom: 1;"><div id="frame-cover" style=" width:310px; height:453px; left:50%;margin-left: -155px; position: fixed; line-height: 0; max-height: calc(100% - 40px);"><iframe sandbox="allow-forms allow-scripts allow-same-origin allow-popups" src="'+t.url+'" width="310" height="420" style="opacity: .98; width:310px; height:420px;  max-height: 100%;" frameborder="0"></iframe><a href="#" id="frame-close"><img style=" display: block; margin: 0 auto; width: 50px; height: 35px; cursor: pointer;" src="'+t.closeIcon+'" alt=""/></a></div></div>',
  t.iconSize=function(e){ return "big" === e ? t.bigIcon : t.smallIcon },
  t.addButton=function(){ document.body.insertAdjacentHTML("afterbegin",t.btn()) },
  t.openFrame=function(e){
    e.preventDefault ? e.preventDefault() : e.returnValue=!1,
    t.appended||(
      document.body.insertAdjacentHTML("afterbegin",t.frame),
      t.modalEl=document.getElementById(t.modalID),
      document.getElementById("frame-close").addEventListener("click",t.closeFrame),
      t.appended=!0
    ),
    t.modalEl.style.display="block",
    document.body.style.overflow="hidden",
    t.frameOpenerEl.style.opacity=0
  },
  t.closeFrame=function(e){
    e.preventDefault?e.preventDefault():e.returnValue=!1,
    document.body.removeAttribute("style"),
    t.modalEl.style.display="none",
    t.frameOpenerEl.style.opacity=1
  },
  t.init=function(){
    t.elementId?t.frameOpenerEl=document.querySelector(t.elementId):(
      t.addButton(),
      t.frameOpenerEl=document.getElementById("frame-opener")
    ),
    t.frameOpenerEl.addEventListener("click",t.openFrame)
}
};


includes/class-wrss-widget.php
<?php

/**
 * Add a widget for plugin drop in for the themes etc
 *
 * @since    0.0.2
 */
class WRSS_Widget extends WP_Widget {

	// Main constructor
	public function __construct() {
		parent::__construct(
			'wrss_widget',
			__( 'Women\'s Refuge Shielded Site Widget', 'WRSS' ),
			array(
				'customize_selective_refresh' => true,
				'classname'                   => 'security', // This gives us a hatched shield default logo in admin tool.
			)
		);
	}

	/**
	 * The widget form (for the backend )
	 *
	 * @since 0.0.2
	 */
	public function form( $instance ) {

		// Set widget defaults
		$defaults = wrss_default_args();

		// Parse current settings with defaults
		extract( wp_parse_args( (array) $instance, $defaults ) );

		echo '<p>';
		echo '	<label for="' . esc_attr( $this->get_field_id( 'icon_size' ) ) . '">' . __( 'Size', 'WRSS' ) . '</label>' . "\n";
		echo '	<select class="widefat" id="' . esc_attr( $this->get_field_id( 'icon_size' ) ) . '" name= "' . esc_attr( $this->get_field_name( 'icon_size' ) ) . '">' . "\n";
		echo '		<option value="large"' . ( 'large' === $icon_size ? ' selected="selected"' : '' ) . '>' . __( 'Large', 'WRSS' ) . '</option>' . "\n";
		echo '		<option value="small"' . ( 'small' === $icon_size ? ' selected="selected"' : '' ) . '>' . __( 'Small', 'WRSS' ) . '</option>' . "\n";
		echo '		<option value="button"' . ( 'button' === $icon_size ? ' selected="selected"' : '' ) . '>' . __( 'Button', 'WRSS' ) . '</option>' . "\n";
		echo '	</select>' . "\n";
		echo '</p>';

		echo '<p>';
		echo '	<label for="' . esc_attr( $this->get_field_id( 'modal_id' ) ) . '">' . __( 'Modal ID', 'WRSS' ) . '</label>' . "\n";
		echo '	<input class="widefat" id="' . esc_attr( $this->get_field_id( 'modal_id' ) ) . '" name="' . esc_attr( $this->get_field_name( 'modal_id' ) ) . '" type="text" value="' . esc_attr( $modal_id ) . '" />' . "\n";
		echo '</p>';

		echo '<p>';
		echo '	<label for="' . esc_attr( $this->get_field_id( 'element_id' ) ) . '">' . __( 'Element ID', 'WRSS' ) . '</label>' . "\n";
		echo '	<input class="widefat" id="' . esc_attr( $this->get_field_id( 'element_id' ) ) . '" name="' . esc_attr( $this->get_field_name( 'element_id' ) ) . '" type="text" value="' . esc_attr( $element_id ) . '" />' . "\n";
		echo '</p>';

	}

	/**
	 * Update widget settings
	 *
	 * @since 0.0.2
	 */
	public function update( $new_instance, $old_instance ) {
		$instance = $old_instance;

		$defaults = wrss_default_args();

		foreach ( $defaults as $name => $val ) {
			$instance[ $name ] = isset( $new_instance[ $name ] ) ? sanitize_text_field( $new_instance[ $name ] ) : $val;
		}

		return $instance;
	}

	/**
	 * Display the widget
	 *
	 * @since 0.0.2
	 */
	public function widget( $args, $instance ) {
		// WordPress core before_widget hook (always include )
		echo $args['before_widget'];

		$instance = wp_parse_args( $instance, wrss_default_args() );

		echo do_shortcode( '[womens_refuge_shield icon_size="' . $instance['icon_size'] . '" modal_id="' . $instance['modal_id'] . '" element_id="' . $instance['element_id'] . '"]' );

		// WordPress core after_widget hook (always include )
		echo $args['after_widget'];
	}

}


/**
 * Register the widget so we can use it in the editors
 *
 * @since    0.0.2
 */
function register_wrss_widget() {
	register_widget( 'WRSS_Widget' );
}
add_action( 'widgets_init', 'register_wrss_widget' );


/**
 * Tiny CSS file - purely to apply the solid shield logo to the Widget in the
 * editor tool.
 *
 * @since    0.0.2
 */
function wrss_enqueue_admin_style( $hook ) {
	wp_register_style( 'wrss_admin_css', WRSS_URL . '/css/admin.css', false, WRSS_VERSION );
	wp_enqueue_style( 'wrss_admin_css' );
}
add_action( 'admin_enqueue_scripts', 'wrss_enqueue_admin_style' );


includes/class-wrss_widget.php
<?php

  /**
   * Add a widget for plugin drop in for the themes etc
   *
   * @since    0.0.2
   */
  class WRSS_Widget extends WP_Widget {

  	// Main constructor
  	public function __construct() {
  		parent::__construct(
  			'WRSS_widget',
  			__( 'Women\'s Refuge Shielded Site Widget', 'WRSS' ),
  			array(
  				'customize_selective_refresh' => true,
  				'classname' => 'security' // This gives us a hatched shield default logo in admin tool.
  			)
  		);
  	}

  	/**
     * The widget form (for the backend )
     *
     * @since 0.0.2
     */
  	public function form( $instance ) {

  		// Set widget defaults
  		$defaults = WRSS_default_args();

  		// Parse current settings with defaults
  		extract( wp_parse_args( ( array ) $instance, $defaults ) );

  		echo '<p>';
  		echo '	<label for="'.esc_attr( $this->get_field_id( 'icon_size' ) ).'">'.__('Size','WRSS').'</label>'."\n";
  		echo '	<select class="widefat" id="'.esc_attr( $this->get_field_id( 'icon_size' ) ).'" name= "'.esc_attr( $this->get_field_name( 'icon_size' ) ).'">'."\n";
  		echo '		<option value="large"'.( $icon_size == 'large'?' selected="selected"':'').'>'.__('Large', 'WRSS').'</option>'."\n";
  		echo '		<option value="small"'.( $icon_size == 'small'?' selected="selected"':'').'>'.__('Small', 'WRSS').'</option>'."\n";
  		echo '		<option value="button"'.( $icon_size == 'button'?' selected="selected"':'').'>'.__('Button', 'WRSS').'</option>'."\n";
  		echo '	</select>'."\n";
  		echo '</p>';

  		echo '<p>';
  		echo '	<label for="'.esc_attr( $this->get_field_id( 'modal_id' ) ).'">'.__('Modal ID','WRSS').'</label>'."\n";
  		echo '	<input class="widefat" id="'.esc_attr( $this->get_field_id( 'modal_id' ) ).'" name="'.esc_attr( $this->get_field_name( 'modal_id' ) ).'" type="text" value="'.esc_attr( $modal_id ).'" />'."\n";
  		echo '</p>';

  		echo '<p>';
  		echo '	<label for="'.esc_attr( $this->get_field_id( 'element_id' ) ).'">'.__('Element ID','WRSS').'</label>'."\n";
  		echo '	<input class="widefat" id="'.esc_attr( $this->get_field_id( 'element_id' ) ).'" name="'.esc_attr( $this->get_field_name( 'element_id' ) ).'" type="text" value="'.esc_attr( $element_id ).'" />'."\n";
  		echo '</p>';

  	}

  	/**
     * Update widget settings
     *
     * @since 0.0.2
     */
  	public function update( $new_instance, $old_instance ) {
  		$instance = $old_instance;

  		$defaults = WRSS_default_args();

  		foreach( $defaults as $name => $val ){
  			$instance[ $name ] = isset( $new_instance[$name] ) ? sanitize_text_field( $new_instance[$name] ) : $val;
  		}

  		return $instance;
  	}

  	/**
     * Display the widget
     *
     * @since 0.0.2
     */
  	public function widget( $args, $instance ) {
  		// WordPress core before_widget hook (always include )
  		echo $args['before_widget'];

  		$instance = wp_parse_args( $instance, WRSS_default_args() );

  		echo do_shortcode( '[womens_refuge_shield icon_size="'.$instance['icon_size'].'" modal_id="'.$instance['modal_id'].'" element_id="'.$instance['element_id'].'"]');

  		// WordPress core after_widget hook (always include )
  		echo $args['after_widget'];
  	}

  }


  /**
   * Register the widget so we can use it in the editors
   *
   * @since    0.0.2
   */
  function register_WRSS_widget() {
  	register_widget( 'WRSS_Widget' );
  }
  add_action( 'widgets_init', 'register_WRSS_widget' );


  /**
   * Tiny CSS file - purely to apply the solid shield logo to the Widget in the
   * editor tool.
   *
   * @since    0.0.2
   */
  function WRSS_enqueue_admin_style( $hook ) {
  	wp_register_style( 'WRSS_admin_css', WRSS_URL . '/css/admin.css', false, WRSS_VERSION );
    wp_enqueue_style( 'WRSS_admin_css' );
  }
  add_action( 'admin_enqueue_scripts', 'WRSS_enqueue_admin_style' );


womens_refuge_shielded_site.php
<?php

/**
 * Women's Refuge Shielded Site
 *
 * @link              https://github.com/PhilTanner/womens_refuge_shielded_site
 * @since             0.0.1
 * @package           WRSS
 * @copyright         2021 Phil Tanner
 *
 * @wordpress-plugin
 * Plugin Name:       Women's Refuge Shielded Site
 * Plugin URI:        https://github.com/PhilTanner/womens_refuge_shielded_site
 * Description:       Add the NZ Women's Refuge Shielded Site button to your site using the [womens_refuge_shield] shortcode or widget.
 * Version:           1.0.3
 * Author:            Phil Tanner
 * Author URI:        https://twitter.com/Phil_Tanner
 * License:           Apache-2.0
 * License URI:       https://www.apache.org/licenses/LICENSE-2.0.html
 * Text Domain:       WRSS
 * Domain Path:       /languages
 *
 *
 *    Copyright 2021 Phil Tanner
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 */

// If this file is called directly, abort.
if ( ! defined( 'WPINC' ) ) {
	die;
}

/**
 * Current plugin version using SemVer - https://semver.org
 *
 * @since    0.0.1
 */
$plugin_data = get_file_data(__FILE__, array('Version' => 'Version') );
define( 'WRSS_VERSION',     $plugin_data['Version'] );

/**
 * Couple of quick-access values to working out our root dir, both file system & URL access.
 *
 * @since    0.0.1
 */
define( 'WRSS_PATH',   __DIR__.DIRECTORY_SEPARATOR );
define( 'WRSS_URL',    plugins_url( '/', __FILE__ ) );

/**
 * Allow not just a shortcode, but a widget to be used.
 *
 * @since    0.0.2
 */
require_once( 'includes'.DIRECTORY_SEPARATOR.'class-wrss_widget.php' );

/**
 * Setup our default arguments so we can refer to them from all over the place.
 *
 * @since    0.0.2
 */
function WRSS_default_args(){
	return array(
		'icon_size'  => 'large', // Accepts 'large','small','button'
		'modal_id'   => 'modal',
		'element_id' => 'shielded-logo',
	);
}

/**
 * Handle our shortcode requests
 *
 * @since    0.0.1
 */
function WRSS_shield( $atts = array(), $content = null, $tag = '' ){

	// Default our arguments
	$atts = shortcode_atts(
		WRSS_default_args(),
		$atts,
		$tag
	);

	// We're only going to accept 'large'/'small'/'button'
	switch( $atts['icon_size'] ){
		case 'button':
			$w = '60';
			$h = '60';
			break;
		case 'small':
			$w = '60';
			$h = '61';
			break;
		// We're going to fall back to this one, which is our default if nothing assigned anyway.
		default:
			$atts['icon_size'] = 'large';
			$w = '60';
			$h = '79.5';
			break;
	}

	// Generate our HTML
	$output = '<div class="wrss_container">'."\n";
	$output .= '	<img alt="'.esc_attr( __('Women\'s Refuge Shielded Site', 'WRSS' ) ).'" '."\n";
	$output .= '		id="'.esc_attr($atts['element_id']).'" '."\n";
	$output .= '		src="'.WRSS_URL.'/img/'.$atts['icon_size'].'-logo.png" '."\n";
	$output .= '		height="'.$h.'" '."\n";
	$output .= '		width="'.$w.'" '."\n";
	$output .= '		style="cursor: pointer; margin: 0px auto; display: inherit;" />'."\n";

	$output .= '	<script>'."\n";
	$output .= '		(function () {'."\n";
	$output .= '			window.onload = function(){'."\n";
	$output .= '				var frameName_'.str_replace('-','_',esc_attr($atts['element_id'])).' = new ds07o6pcmkorn({'."\n";
	$output .= '					openElementId: "#'.esc_attr($atts['element_id']).'",'."\n";
	$output .= '					modalID:       "'.esc_attr($atts['modal_id']).'",'."\n";
	$output .= '				});'."\n";
	$output .= '				frameName_'.str_replace('-','_',esc_attr($atts['element_id'])).'.init();'."\n";
	$output .= '			}'."\n";
	$output .= '		})();'."\n";
	$output .= '	</script>'."\n";


	$output .= '</div>'."\n";

	// Pass it back for inclusion
	return $output;
}
add_shortcode( 'womens_refuge_shield', 'WRSS_shield' );


/**
 * WordPress approval process wanted the script moved to enqueue - so we're including
 * here to meet those requirements - it means we're ALWAYS including it, even if
 * the icon isn't used - we'll just have to assume that if they've got the plugin,
 * and it's activated, they are using it somewhere.
 *
 * @since    1.0.1
 */
function wrss_embed_external_script() {
	wp_enqueue_script( 'WRSS_external_embed', 'https://staticcdn.co.nz/embed/embed.js'  );
}
add_action( 'wp_enqueue_scripts', 'WRSS_embed_external_script' );

/**
 * Add some additional links underneath our plugin description,
 * linking back to the orginal websites for more information
 *
 * @since    1.0.1
 */
function WRSS_defer_embed_script($tag, $handle) {
	// We want to defer our embed, as it's unlikely the user will ever click the
	// button before the page has loaded, and there's no point slowing the rendering
	// for it.
	if( $handle == 'WRSS_external_embed' ){
		return str_ireplace(' src', ' defer src', $tag);
	}
	// Not our tag, so we'll return it as we found it.
	return $tag;
}
add_filter('script_loader_tag', 'WRSS_defer_embed_script', 10, 2);

/**
 * Add some additional links underneath our plugin description,
 * linking back to the orginal websites for more information
 *
 * @since    0.0.1
 */
function WRSS_plugin_links( $links_array, $plugin_file_name, $plugin_data, $status ){
	if( strpos( $plugin_file_name, basename(__FILE__) ) ) {
		$links_array[] = '<a href="https://shielded.co.nz/" target="_blank" class="dashicons-before dashicons-external">'.__('The Shielded Site Project','WRSS').'</a>';
		$links_array[] = '<a href="https://womensrefuge.org.nz/" target="_blank" class="dashicons-before dashicons-external">'.__('Women\'s Refuge NZ', 'WRSS').'</a>';
	}
	return $links_array;
}
add_filter( 'plugin_row_meta', 'WRSS_plugin_links', 10, 4 );

/**
 * Add a Donation link to the plugin deactivate area.
 * Donating to the Refuge, not me.
 *
 * @since    0.0.1
 */
function WRSS_action_links( $links_array, $plugin_file_name, $plugin_data, $status ){
	if( strpos( $plugin_file_name, basename(__FILE__) ) ) {
		$links_array[] = '<a href="https://womensrefuge.org.nz/make-a-donation/" class="dashicons-before dashicons-money-alt">'.__('Donate to Women\'s Refuge', 'WRSS').'</a>';
	}
	return $links_array;
}
add_filter( 'plugin_action_links', 'WRSS_action_links', 10, 4 );


womens-refuge-shielded-site.php
<?php

/**
 * Women's Refuge Shielded Site
 *
 * @link              https://github.com/PhilTanner/womens_refuge_shielded_site
 * @since             0.0.1
 * @package           WRSS
 * @copyright         2021 Phil Tanner
 *
 * @wordpress-plugin
 * Plugin Name:       Women's Refuge Shielded Site
 * Plugin URI:        https://github.com/PhilTanner/womens_refuge_shielded_site
 * Description:       Add the NZ Women's Refuge Shielded Site button to your site using the [womens_refuge_shield] shortcode or widget.
 * Version:           1.0.4
 * Author:            Phil Tanner
 * Author URI:        https://twitter.com/Phil_Tanner
 * License:           Apache-2.0
 * License URI:       https://www.apache.org/licenses/LICENSE-2.0.html
 * Text Domain:       WRSS
 * Domain Path:       /languages
 *
 *
 *    Copyright 2021 Phil Tanner
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 */

// If this file is called directly, abort.
if ( ! defined( 'WPINC' ) ) {
	die;
}

/**
 * Current plugin version using SemVer - https://semver.org
 *
 * @since    0.0.1
 */
$plugin_data = get_file_data( __FILE__, array( 'Version' => 'Version' ) );
define( 'WRSS_VERSION', $plugin_data['Version'] );

/**
 * Couple of quick-access values to working out our root dir, both file system & URL access.
 *
 * @since    0.0.1
 */
define( 'WRSS_PATH', __DIR__ . DIRECTORY_SEPARATOR );
define( 'WRSS_URL', plugins_url( '/', __FILE__ ) );

/**
 * Allow not just a shortcode, but a widget to be used.
 *
 * @since    0.0.2
 */
require_once 'includes' . DIRECTORY_SEPARATOR . 'class-wrss-widget.php';

/**
 * Setup our default arguments so we can refer to them from all over the place.
 *
 * @since    0.0.2
 */
function wrss_default_args() {
	return array(
		'icon_size'  => 'large', // Accepts 'large','small','button'
		'modal_id'   => 'modal',
		'element_id' => 'shielded-logo',
	);
}

/**
 * Handle our shortcode requests
 *
 * @since    0.0.1
 */
function wrss_shield( $atts = array(), $content = null, $tag = '' ) {

	// Default our arguments
	$atts = shortcode_atts(
		wrss_default_args(),
		$atts,
		$tag
	);

	// We're only going to accept 'large'/'small'/'button'
	switch ( $atts['icon_size'] ) {
		case 'button':
			$w = '60';
			$h = '60';
			break;
		case 'small':
			$w = '60';
			$h = '61';
			break;
		// We're going to fall back to this one, which is our default if nothing assigned anyway.
		default:
			$atts['icon_size'] = 'large';
			$w                 = '60';
			$h                 = '79.5';
			break;
	}

	// Generate our HTML
	$output  = '<div class="wrss_container">' . "\n";
	$output .= '	<img alt="' . esc_attr( __( 'Women\'s Refuge Shielded Site', 'WRSS' ) ) . '" ' . "\n";
	$output .= '		id="' . esc_attr( $atts['element_id'] ) . '" ' . "\n";
	$output .= '		src="' . WRSS_URL . '/img/' . $atts['icon_size'] . '-logo.png" ' . "\n";
	$output .= '		height="' . $h . '" ' . "\n";
	$output .= '		width="' . $w . '" ' . "\n";
	$output .= '		style="cursor: pointer; margin: 0px auto; display: inherit;" />' . "\n";

	$output .= '	<script>' . "\n";
	$output .= '		(function () {' . "\n";
	$output .= '			window.onload = function(){' . "\n";
	$output .= '				var frameName_' . str_replace( '-', '_', esc_attr( $atts['element_id'] ) ) . ' = new ds07o6pcmkorn({' . "\n";
	$output .= '					openElementId: "#' . esc_attr( $atts['element_id'] ) . '",' . "\n";
	$output .= '					modalID:       "' . esc_attr( $atts['modal_id'] ) . '",' . "\n";
	$output .= '				});' . "\n";
	$output .= '				frameName_' . str_replace( '-', '_', esc_attr( $atts['element_id'] ) ) . '.init();' . "\n";
	$output .= '			}' . "\n";
	$output .= '		})();' . "\n";
	$output .= '	</script>' . "\n";

	$output .= '</div>' . "\n";

	// Pass it back for inclusion
	return $output;
}
add_shortcode( 'womens_refuge_shield', 'wrss_shield' );


/**
 * WordPress approval process wanted the script moved to enqueue - so we're including
 * here to meet those requirements - it means we're ALWAYS including it, even if
 * the icon isn't used - we'll just have to assume that if they've got the plugin,
 * and it's activated, they are using it somewhere.
 *
 * @since    1.0.1
 */
function wrss_embed_external_script() {
	wp_enqueue_script( 'wrss_external_embed', 'https://staticcdn.co.nz/embed/embed.js' );
}
add_action( 'wp_enqueue_scripts', 'wrss_embed_external_script' );

/**
 * Add some additional links underneath our plugin description,
 * linking back to the orginal websites for more information
 *
 * @since    1.0.1
 */
function wrss_defer_embed_script( $tag, $handle ) {
	// We want to defer our embed, as it's unlikely the user will ever click the
	// button before the page has loaded, and there's no point slowing the rendering
	// for it.
	if ( 'wrss_external_embed' === $handle ) {
		return str_ireplace( ' src', ' defer src', $tag );
	}
	// Not our tag, so we'll return it as we found it.
	return $tag;
}
add_filter( 'script_loader_tag', 'wrss_defer_embed_script', 10, 2 );

/**
 * Add some additional links underneath our plugin description,
 * linking back to the orginal websites for more information
 *
 * @since    0.0.1
 */
function wrss_plugin_links( $links_array, $plugin_file_name, $plugin_data, $status ) {
	if ( strpos( $plugin_file_name, basename( __FILE__ ) ) ) {
		$links_array[] = '<a href="https://shielded.co.nz/" target="_blank" class="dashicons-before dashicons-external">' . __( 'The Shielded Site Project', 'WRSS' ) . '</a>';
		$links_array[] = '<a href="https://womensrefuge.org.nz/" target="_blank" class="dashicons-before dashicons-external">' . __( 'Women\'s Refuge NZ', 'WRSS' ) . '</a>';
	}
	return $links_array;
}
add_filter( 'plugin_row_meta', 'wrss_plugin_links', 10, 4 );

/**
 * Add a Donation link to the plugin deactivate area.
 * Donating to the Refuge, not me.
 *
 * @since    0.0.1
 */
function wrss_action_links( $links_array, $plugin_file_name, $plugin_data, $status ) {
	if ( strpos( $plugin_file_name, basename( __FILE__ ) ) ) {
		$links_array[] = '<a href="https://womensrefuge.org.nz/make-a-donation/" class="dashicons-before dashicons-money-alt">' . __( 'Donate to Women\'s Refuge', 'WRSS' ) . '</a>';
	}
	return $links_array;
}
add_filter( 'plugin_action_links', 'wrss_action_links', 10, 4 );
